### DISCUSSION TITLE
Publish major version for aws-sigv4

### DISCUSSION THREAD

**Discussion by steveh** [aws/aws-sdk-ruby#3145] [General]
RE: https://github.com/aws/aws-sdk-ruby/issues/3027

It's probably too late now unless you yank the published version, but in my opinion the change to add `content-type` should have been a major version upgrade for aws-sigv4. This change was not backwards compatible with existing versions of aws-sdk-s3, and is not obvious from the changelog of either aws-sdk-s3 or aws-sigv4.

**Comment by mullermp** [comment]
Thank you for the feedback. Bumping the sigv4 major version would have caused more conflict because of how it's depended on for aws-sdk-core and the service gems, and common usage of more than one gem, would have caused a conflict of two major versions in bundle resolution.
